frontend/public/components/utils/tile-view-page.jsx

```diff
@@ -635,6 +671,14 @@ export class TileViewPage extends React.Component {
     const updatedFilters = updateActiveFilters(activeFilters, filterType, id, value);
 
     this.updateMountedState(this.getUpdatedState(categories, selectedCategoryId, updatedFilters));
+
+    if (this.props.storeFilterKey && updatedFilters) {
+      const storeFilters = {};
+      _.each(this.props.filterRetentionPreference, (filterGroup) => {
```

Comment:
We shouldn't look to write lazy code just because lodash exists. This is an optional property we don't need to be saving filters on if there is no preference.

Reply:
unable to understand this :(

Reply:
```
if (this.props.storeFilterKey && updatedFilters) {
      const storeFilters = {};
      _.each(this.props.filterRetentionPreference, (filterGroup) => {
```

If we have a `storeFilterKey` and `updatedFilters` loop over `filterRetentionPreference` (which could be missing). Lodash saves us from this NPE issue... but you just then save an empty object into local storage.

Add `filterRetentionPreference` to your condition as you're relying on its value without checking first. It could also be an empty array which has the same problem.

Reply:
Updated

Reply:
@abhinandan13jan I don't see this updated - forgot to push?

Reply:
not sure, I guess so.. updated